app/src/main/java/io/crate/plugin/PluginLoader.java

```diff
@@ -282,9 +284,10 @@ Settings additionalSettings() {
 
 
     private void checkJarHell(URL url) throws Exception {
-        final List<URL> loadedJars = new ArrayList<>(Arrays.asList(JarHell.parseClassPath()));
-        loadedJars.addAll(jarsToLoad);
-        loadedJars.add(url);
-        JarHell.checkJarHell(loadedJars.toArray(new URL[0]));
+        Set<URL> jarHellSet = new HashSet<>();
```

Comment:
Doesn't deduplicating the URLs kinda work-around of one of the purposes of the JarHell check?